lib/Dialect/FIRRTL/Transforms/ModuleInliner.cpp

```diff
@@ -441,6 +476,42 @@ class Inliner {
 };
 } // namespace
 
+/// If this operation or any child operation has a name, add the prefix to that
+/// operation's name.  If the operation has any inner symbols, make sure that
+/// these are unique in the namespace.
+void Inliner::rename(StringRef prefix, Operation *op,
+                     ModuleNamespace &moduleNamespace) {
+  llvm::TypeSwitch<Operation *>(op)
+      .Case<CombMemOp, InstanceOp, MemOp, MemoryPortOp, NodeOp, RegOp,
+            RegResetOp, SeqMemOp, WireOp>([&](auto op) {
+        op.nameAttr(
+            StringAttr::get(op.getContext(), (prefix + op.name()).str()));
+      });
+
+  // If the operation has an inner symbol, ensure that it is unique.  Record
+  // renames for any NLAs that this participates in if the symbol was renamed.
+  if (auto sym = op->getAttrOfType<StringAttr>("inner_sym")) {
+    auto newSym = moduleNamespace.newName(sym.getValue());
+    if (newSym != sym.getValue()) {
+      auto newSymAttr = StringAttr::get(op->getContext(), newSym);
+      op->setAttr("inner_sym", newSymAttr);
+      for (Annotation anno : AnnotationSet(op)) {
+        auto sym = anno.getMember<FlatSymbolRefAttr>("circt.nonlocal");
+        if (!sym)
+          continue;
+        nlaMap[sym.getAttr()].setInnerSym(
+            moduleNamespace.module.moduleNameAttr(), newSymAttr);
+      }
+    }
+  }
+
+  // Recursively rename any child operations.
+  for (auto &region : op->getRegions())
```

Comment:
Seems like this could be done during clone/splice rather than as a second pass over the IR.

Reply:
I guess the builder doesn't have the hooks.